src/rewriter/rw_parser.py

```diff
@@ -266,25 +168,50 @@ def var_list(self):
                  lambda s, l, t: self.var_decl_action(t[0], t[1], t[2:])))
         return (pp.Suppress('(') + pp.ZeroOrMore(decl) + pp.Suppress(')'))
 
+    def def_list(self):
+        d = pp.Suppress((pp.Suppress('(') + self.symbol() + self.expr()
+                         + pp.Suppress(')'))
+                .setParseAction(lambda s, l, t: self.def_decl_action(t[0], t[1])))
+        return pp.Optional(pp.Suppress('(') + pp.Suppress(pp.Keyword('def')) +
+                           pp.ZeroOrMore(d) + pp.Suppress(')'))
+
     def rule_action(self, name, cond, lhs, rhs, is_fixed_point, rhs_context):
         bvars = self.symbols.symbols.values()
         self.symbols.pop()
         return Rule(name, bvars, cond, lhs, rhs, is_fixed_point, rhs_context)
 
     def parse_rules(self, s):
+        def rule_action(s, l, t):
+            # t = [key, args, match, target]
+            assert len(t) == 4
+            return self.rule_action(t[1], CBool(True), t[2], t[3], False, None)
```

Comment:
How about splitting t into actual variables:

```suggestion
        def rule_action(s, l, t):
            assert len(t) == 4
            key, args, match, target = t
            return self.rule_action(args, CBool(True), match, target, False, None)
```


```
assert len(t) == 4 or len(t) == 5
key, args, match, target, cond = t + [None]
```

Reply:
fixed. need to convert `t` to a list to prevent the build failure in CI